src/core/binding.c

```diff
@@ -426,7 +431,20 @@ QuicBindingGetListener(
 
     CxPlatDispatchRwLockReleaseShared(&Binding->RwLock);
 
+    if (FailedAddrMatch) {
+        QuicTraceEvent(
+            ConnNoListenerIp,
+            "[conn][%p] No Listener for IP address: %!ADDR!",
+            Connection,
+            CLOG_BYTEARRAY(sizeof(*Addr), Addr));
+    }
+
     if (FailedAlpnMatch) {
```

Comment:
```suggestion
    } else if (FailedAlpnMatch) {
```
It shouldn't be possible to be both.